timelock-impl/src/test/java/com/palantir/atlasdb/timelock/TimelockNamespacesTest.java

```diff
@@ -154,6 +155,29 @@ public void pathsForTimelockAndLockWatchServicesAreNotValid() {
         assertThat(TimelockNamespaces.IS_VALID_NAME.test("lwbleh")).isTrue();
     }
 
+    @Test
+    public void invalidationDelegatesClosure() {
+        // This is required to ensure we get different mock objects on each invocation (well, each of the first two).
+        when(serviceFactory.apply(any()))
+                .thenReturn(mock(TimeLockServices.class))
+                .thenReturn(mock(TimeLockServices.class));
+
+        String client = uniqueClient();
+        TimeLockServices services = namespaces.get(client);
+        namespaces.invalidateResourcesForClient(client);
+        verify(services).close();
+
+        services = namespaces.get(client);
+        namespaces.invalidateResourcesForClient(client);
+        verify(services).close();
```

Comment:
can we add an assertion that 2 service objects were different?

Reply:
Yep, will do.